src/user_management/user.model.js

```diff
@@ -1,11 +1,40 @@
+import bcrypt from 'bcrypt'
+
 export default (sequelize, DataTypes) => {
   const User = sequelize.define('user', {
     id: {
       type: DataTypes.UUID,
       primaryKey: true,
       defaultValue: DataTypes.UUIDV4
+    },
+    firstName: {
+      type: DataTypes.STRING,
+      allowNull: false
+    },
+    lastName: {
+      type: DataTypes.STRING,
+      allowNull: false
+    },
+    email: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      unique: true
+    },
+    password: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      
     }
   })
 
+  User.addHook('beforeCreate', async (user, options) => {
+    const salt = await bcrypt.genSalt(10)
+    user.password = await bcrypt.hash(user.password, salt)
+  })
+
+  User.prototype.validatePass = async function (password) {
+    return await bcrypt.compare(password, this.password)
+  }
```

Comment:
- If you're returning the promise, there's no need for an `await`.
- Try not to use abbreviations for names.

```suggestion
  User.prototype.validatePassword = function (password) {
    return bcrypt.compare(password, this.password)
  }
```